src/main/java/com/kuljava/swiatwsi/services/RawMaterialsService.java

```diff
@@ -0,0 +1,32 @@
+package com.kuljava.swiatwsi.services;
+
+import com.kuljava.swiatwsi.rawmaterials.RawMaterials;
+import com.kuljava.swiatwsi.rawmaterials.RawMaterialsRepository;
+import com.kuljava.swiatwsi.world.Village;
+import com.kuljava.swiatwsi.world.VillageRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+public class RawMaterialsService {
+  @Autowired VillageRepository villageRepository;
+  @Autowired RawMaterialsRepository rawMaterialsRepository;
+
+  public Optional<RawMaterials> getRawMaterialsByVillageId(Long id) {
+    Optional<Village> village = villageRepository.findById(id);
+    return village.isEmpty()
+        ? Optional.empty()
+        : Optional.ofNullable(village.get().getRawMaterials());
+  }
+
+  public void createRawMaterialsForVillage(Village village) {
+    if (village != null) {
```

Comment:
Try to replace this with optional.

Reply:
The method return type?

Reply:
No, this null check. Optional is more suitable for this job :) smth like Optional.ofNullable.map(this::doSomeJob).map(this::doAnotherJob).ifPresent(this::doFinalJob);

Reply:
Sometimes Optional is also suitable for working on data without returning it. :)

Reply:
Niestety nie ogarniam tego na razie więc musi zostać tak jak jest dopóki się nie douczę, może kiedy to zmienię :)